python/interpret_text/experimental/introspective_rationale/introspective_rationale_explainer.py

```diff
@@ -185,46 +185,47 @@ def _load_defaults_model_config_dict(self) -> Dict:
             config = {}
         return config
 
-    def explain_local(self, text: str, **kwargs) -> _create_local_explanation:
+    def explain_local(self, X, y=None, name=None) -> LocalExplanation:
         """ Create a local explanation for a given text
-        :param text: A segment of text
-        :type text: str
-        :param kwargs:
-                preprocessor: an intialized preprocessor to tokenize the
-                given text with .preprocess() and .decode_single() methods
-                preprocessor: Ex. GlovePreprocessor or BertPreprocessor
-                hard_importances: whether to generate "hard" important/ non-important rationales
-                or float rationale scores, defaults to True
-                hard_importances: bool, optional
+        :param X: String to be explained.
+        :type X: str
+        :param y: The ground truth label for the sentence
+        :type y: string
+        :param name: a name for saving the explanation, currently ignored
+        :type str
         :return: local explanation object
         :rtype: DynamicLocalExplanation
         """
+        X = _validate_X(X)
+
         model_args = self.model_config
         df_dummy_label = pd.DataFrame.from_dict({"labels": [0]})
         df_sentence = pd.concat(
-            [df_dummy_label, self.preprocessor.preprocess([text.lower()])],
+            [df_dummy_label, self.preprocessor.preprocess([X.lower()])],
             axis=1
         )
         batch_dict = generate_data(df_sentence, self.model_config.cuda)
         x = batch_dict["x"]
         m = batch_dict["m"]
         predict_dict = self.predict(df_sentence)
         zs = predict_dict["rationale"]
+
+        # The not hard_importance condition was implied, ids is undefined otherwise
```

Comment:
it looks like this code was indented incorrectly, as it's unrelated to computing the "hard importances", which should only be related to the zs:

```
        # generate human-readable tokens (individual words)
        seq_len = int(m.sum().item())
        ids = x[:seq_len][0]
```

Reply:
True, I'll move the comment down. nice catch